### PR TITLE
Prevent GitLab configuration from poking at internal IP addresses

### PR DESCRIPTION
**Pull request by mattrobenolt**
@getsentry/security @macqueen 

---

This change is [Reviewable](https://reviewable.io/reviews/getsentry/sentry-gitlab/1)

### CODE REVIEW

**Comment by dcramer** [comment]
Assuming GitLab SDK accepts redirects, its very easy to bypass this protection through that

**Comment by mattrobenolt** [comment]
Hm, that's a good point. Yeah, so the only way to get around this is to implement our own little SDK that does this one thing. I can do that.

**Comment by macqueen** [comment]
Reviewed 3 of 3 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry-gitlab/1)_

**Comment by dcramer** [comment]
We need to drop python-gitlab as its incompatible with the current version of GitLab's API (it asserts a 201 on issue creation, and it may be a 200 in some versions). I think to achieve that we should just priority getting this in.

File: src/sentry_gitlab/plugin.py [hunk 1]
```diff
@@ -65,39 +65,54 @@ def get_new_issue_title(self, **kwargs):
...
-            raise forms.ValidationError(_('Unauthorized: Invalid Private Token: %s') % (e,))
+            response = session.get(
+                url='%s/api/v3/projects/%s' % (url, repo_url),
+                headers=headers,
+                allow_redirects=False,
+            )
+            response.raise_for_status()
+
+            project_id = response.json()['id']
+
```
Comment (yamadapc) [inline src/sentry_gitlab/plugin.py]: You don't need this if you URL encode the `owner/slug` stuff

Comment (mattrobenolt) [reply src/sentry_gitlab/plugin.py]: I'll double check this, but I _thought_ you needed to resolve it to an id. If not, that'd be great.

Comment (yamadapc) [reply src/sentry_gitlab/plugin.py]: I just tested:

```
curl -s -XPOST -H "PRIVATE-TOKEN: $GITLAB_TOKEN" "https://gitlab.com/api/v3/projects/$TARGET_REPO/issues" -d "title=test-issue"
```

Comment (mattrobenolt) [reply src/sentry_gitlab/plugin.py]: Yup, confirmed. Not needed. Thanks for pointing it out. :)

Comment (yamadapc) [reply src/sentry_gitlab/plugin.py]: Awesome! No worries!

Thanks for the great work on Sentry.

**Comment by dcramer** [comment]
Reviewed 1 of 3 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/getsentry/sentry-gitlab/1)_

**Pull request merged**